### DISCUSSION TITLE
failure on "every" lxc installation because of locales

### DISCUSSION THREAD

**Discussion by rootgremlin** [tteck/Proxmox#1159] [Help Needed]
i can not use most of the installations because near the end of the scripts it always fails with:

[ERROR] in line 48: exit code 1: while executing command sed -i "/$LANG/ s/\(^# \)//" /etc/locale.gen

the verbose output of one failed attempt:

msg_info "Cleaning up"
$STD apt-get autoremove
$STD apt-get autoclean
msg_ok "Cleaned"'
+ STD=
+ '[' yes == yes ']'
+ echo 'net.ipv6.conf.all.disable_ipv6 = 1'
+ sysctl -p
net.ipv6.conf.all.disable_ipv6 = 1
++ echo '\033[33m'
+ YW='\033[33m'
++ echo '\033[01;31m'
+ RD='\033[01;31m'
++ echo '\033[36m'
+ BL='\033[36m'
++ echo '\033[1;92m'
+ GN='\033[1;92m'
++ echo '\033[m'
+ CL='\033[m'
+ RETRY_NUM=10
+ RETRY_EVERY=3
+ CM='\033[1;92m✓\033[m'
+ CROSS='\033[01;31m✗\033[m'
+ BFR='\r\033[K'
+ HOLD=-
+ set -Eeuo pipefail
+ trap 'error_handler $LINENO "$BASH_COMMAND"' ERR
+ msg_info 'Setting up Container OS'
+ local 'msg=Setting up Container OS'
+ echo -ne ' - \033[33mSetting up Container OS...'
 - Setting up Container OS...+ sed -i '// s/\(^# \)//' /etc/locale.gen
sed: -e expression #1, char 0: no previous regular expression
++ error_handler 47 'sed -i "/$LANG/ s/\(^# \)//" /etc/locale.gen'
++ local exit_code=1
++ local line_number=47
++ local 'command=sed -i "/$LANG/ s/\(^# \)//" /etc/locale.gen'
++ local 'error_message=\033[01;31m[ERROR]\033[m in line \033[01;31m47\033[m: exit code \033[01;31m1\033[m: while executing command \033[33msed -i "/$LANG/ s/\(^# \)//" /etc/locale.gen\033[m'
++ echo -e '\n\033[01;31m[ERROR]\033[m in line \033[01;31m47\033[m: exit code \033[01;31m1\033[m: while executing command \033[33msed -i "/$LANG/ s/\(^# \)//" /etc/locale.gen\033[m\n'

[ERROR] in line 47: exit code 1: while executing command sed -i "/$LANG/ s/\(^# \)//" /etc/locale.gen

+ exit

is there something wrong with my locales config on my proxmox host?  at least i had no problem with it until now.

Yes the command sed -i "/$LANG/ s/\(^# \)//" /etc/locale.gen also fails inside the newly created LXC because $LANG is empty.

Where should it get populated, or what do i have to set for the installer to pick my language up?

**Comment by tteck** [comment]
You can use the `dpkg-reconfigure locales` command to set the system locales.

**Reply by tteck** [reply]
If you are currently using SSH, I would suggest trying to use the Proxmox Shell instead to see if that resolves the issue.

**Reply by rootgremlin** [reply]
> You can use the `dpkg-reconfigure locales` command to set the system locales.

i did!
i have the de_DE locales for my workstation(s) and en_US for servers. 
Without "dpkg-reconfigure  locales" a remote ssh session always complains about missing locales
 example:
 perl: warning: Setting locale failed.
perl: warning: Please check that your locale settings:
	LANGUAGE = (unset),
	LC_ALL = (unset),
	LC_TIME = "de_DE.UTF-8",
	LC_MONETARY = "de_DE.UTF-8",
	LC_ADDRESS = "de_DE.UTF-8",
	LC_TELEPHONE = "de_DE.UTF-8",
	LC_NAME = "de_DE.UTF-8",
	LC_MEASUREMENT = "de_DE.UTF-8",
	LC_IDENTIFICATION = "de_DE.UTF-8",
	LC_NUMERIC = "de_DE.UTF-8",
	LC_PAPER = "de_DE.UTF-8",
	LANG = "C"
    are supported and installed on your system.
 
 
 the problem is, how can i set it inside the "currently" LXC while it is in the process of beeing created?
 
 that was the reason i asked
 > Yes the command sed -i "/$LANG/ s/(^# )//" /etc/locale.gen also fails inside the newly created LXC because $LANG is empty.
> 
> Where should it get populated, or what do i have to set for the installer to pick my language up?

**Reply by rootgremlin** [reply]
> If you are currently using SSH, I would suggest trying to use the Proxmox Shell instead to see if that resolves the issue.

I don's see how a "error free" (which it is)  ssh session to the proxmox host can influence the $LANG variable beeing set while the "new" container is in the process of beeing created?

My Problem is, i don't see, where this line ( sed -i "/$LANG/ s/(^# )//" /etc/locale.gen) comes from, as searching in the script for $LANG did not yield any result

**Reply by tteck** [reply]
Have you tried running the command from the Proxmox Shell?

**Reply by rootgremlin** [reply]
> Have you tried running the command from the Proxmox Shell?

yes i did which yielded the same result....

As i have found out, for whatever reason my $LANG environment variable is unpopulated. when i set the LANG to enything like with " export LANG=en_US " all is well end the setup-script finishes

**Comment by tteck** [comment]
The script gets its information from Proxmox when running the command from the shell.

**Reply by rootgremlin** [reply]
a few of the "old" environment variables:

# env
SHELL=/bin/bash
LANGUAGE=
LC_ADDRESS=de_DE.UTF-8
LC_NAME=de_DE.UTF-8
LC_MONETARY=de_DE.UTF-8
XDG_SESSION_TYPE=tty
LS_OPTIONS=--color=auto
LANG=
LC_PAPER=de_DE.UTF-8
XDG_SESSION_CLASS=user
TERM=xterm-256color
LC_IDENTIFICATION=de_DE.UTF-8
LC_TELEPHONE=de_DE.UTF-8
LC_MEASUREMENT=de_DE.UTF-8
LC_TIME=de_DE.UTF-8
LC_NUMERIC=de_DE.UTF-8
_=/usr/bin/env

with the empty $LANG variable.

After setting it to "anything" non empty with:

export LANG=en_US 

the script completed successfully

So thank you!!
